librz/core/agraph.c

```diff
@@ -4956,14 +4962,30 @@ RZ_API RzAGraph *create_agraph_from_graph(const RzGraph /*<RzGraphNodeInfo *>*/
 			if (!a_neighbour) {
 				goto failure;
 			}
-			rz_agraph_add_edge(result_agraph, a_neighbour, a_node);
+			rz_agraph_add_edge(ag, a_neighbour, a_node);
 		}
 	}
 
 	ht_pp_free(hashmap);
-	return result_agraph;
+	return ag;
 failure:
 	ht_pp_free(hashmap);
-	rz_agraph_free(result_agraph);
+	rz_agraph_free(ag);
 	return NULL;
 }
+
+/**
+ * @brief Create RzAGraph from generic RzGraph with RzGraphNodeInfo as node data
+ *
+ * @param graph <RzGraphNodeInfo>
+ * @return RzAGraph* NULL if failure
+ */
+RZ_API RzAGraph *create_agraph_from_graph(const RzGraph /*<RzGraphNodeInfo *>*/ *graph) {
```

Comment:
The same concern about the API. In fact, looks like only this one should be an API while the previous could be a `static` function.